### PR TITLE
Move LVGL gluecode back into main repository

### PR DESCRIPTION
**Pull request by faxe1008**
Addresses: [https://github.com/zephyrproject-rtos/zephyr/issues/60868](https://github.com/zephyrproject-rtos/zephyr/issues/60868)

The related PR in the module: [https://github.com/zephyrproject-rtos/lvgl/pull/45](https://github.com/zephyrproject-rtos/lvgl/pull/45).

### CODE REVIEW

**Comment by faxe1008** [comment]
@fabiobaltieri I merged the changes from [pull/44](https://github.com/zephyrproject-rtos/lvgl/pull/44) into here.

Building the sample works fine for me, but there are still some CI issues and I have no idea on how I can resolve them:
[CI issue 1](https://github.com/zephyrproject-rtos/zephyr/actions/runs/5812998355/job/15759623873?pr=61300#step:9:26)
[CI issue 2](https://github.com/zephyrproject-rtos/zephyr/actions/runs/5812998280/job/15759623656?pr=61300#step:12:627)

Simply put I need to somehow source the Kconfig from the module but don't know what the right way to do it is, considering I also have code in the zephyr main repo.

**Comment by fabiobaltieri** [comment]
> Simply put I need to somehow source the Kconfig from the module but don't know what the right way to do it is, considering I also have code in the zephyr main repo. Do you have any insights here :^) ?

Yeah so you want the module system to only include the Kconfig when it's there, just drop `kconfig-ext: True` from `module.yml` in the LVGL module PR.

```
diff --git a/zephyr/module.yml b/zephyr/module.yml
index e0e7d676..38ef2dd2 100644
--- a/zephyr/module.yml
+++ b/zephyr/module.yml
@@ -2,5 +2,4 @@
 name: lvgl
 build:
   cmake-ext: True
-  kconfig-ext: True
   kconfig: Kconfig
```

Then the Kconfig that is always there needs to have stub symbols with defined types for everything that is overridden at board level, apply this:

```diff
diff --git a/modules/lvgl/Kconfig b/modules/lvgl/Kconfig
index 43b187abee..9d1b1bc1da 100644
--- a/modules/lvgl/Kconfig
+++ b/modules/lvgl/Kconfig
@@ -11,9 +11,14 @@ config LVGL
 
 if LVGL
 
-source "${ZEPHYR_LVGL_MODULE_DIR}/Kconfig"
+config LV_USE_MONKEY
+       bool
+
+config LV_DPI_DEF
+       int
 
 config LV_CONF_SKIP
+       bool
        default n
 
 config APP_LINK_WITH_LVGL
@@ -25,7 +30,7 @@ config APP_LINK_WITH_LVGL
          issues for 'app'.
 
 config LV_Z_USE_FILESYSTEM
-       bool "Enable file system"
+       bool "LVGL file system support"
        depends on FILE_SYSTEM
        default y if FILE_SYSTEM
        help
```

**Comment by fabiobaltieri** [comment]
By the way, you can run the compliance checks locally with `./scripts/ci/check_compliance.py`, something like `./scripts/ci/check_compliance.py  -c origin/main..`, saves you from pushing and waiting for CI to run it for you.

**Comment by faxe1008** [comment]
Thanks @fabiobaltieri for the hints, apart from a checkpatch false positive everything builds fine now.

**Comment by fabiobaltieri** [comment]
Thanks for taking care of this, looks good, should be a single commit though so if a bisect lands on the first one it builds correctly.

**Comment by faxe1008** [comment]
Squashed the commits down :^)

File: modules/lvgl/lv_conf.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018-2020 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_GUI_LVGL_LV_CONF_H_
```
Comment (fabiobaltieri) [inline modules/lvgl/lv_conf.h:8]: Can you rename the guards? This should now be `ZEPHYR_MODULES_LVGL_LV_CONF_H_`

File: modules/lvgl/lvgl_fs.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2018 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_GUI_LVGL_LVGL_FS_H_
```
Comment (fabiobaltieri) [inline modules/lvgl/lvgl_fs.h:7]: same

File: west.yml [hunk 3]
```diff
@@ -260,7 +260,7 @@ manifest:
       revision: ce57712f3e426bbbb13acaec97b45369f716f43a
       path: modules/lib/loramac-node
     - name: lvgl
-      revision: 5da257f782a8f9c6e265bdc60ebc2a93fdee24de
+      revision: pull/45/head
```
Comment (fabiobaltieri) [inline west.yml:263]: 8a6a2d1d29d17d1e4bdc94c243c146a39d635fdd there you go

**Comment by faxe1008** [comment]
Adjusted all include guards, updated revision :^)

**Review by fabiobaltieri — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
@gmarull @jfischer-no @danieldegrasse any second approval for this? The CI fail is a false positive.

**Review by danieldegrasse — APPROVED** [review]
Thanks for taking this on- looks good to me, tested on an RT1170 with the LVGL benchmark and widgets demos and everything seems functional
